var/spack/repos/builtin/packages/jdk/package.py

```diff
@@ -22,60 +22,167 @@
-    # https://linuxconfig.org/how-to-install-java-se-development-kit-on-debian-linux
-    version('8u172-b11', 'eda2945e8c02b84adbf78f46c37b71c1', curl_options=curl_options,
+    # To add the latest version, go to the homepage listed above,
+    # click "JDK Download", click "Accept License Agreement", right-click the
+    # Linux .tar.gz link, and select Copy Link Address. The checksum can be
+    # found in a link above. The build number can be deciphered from the URL.
+    # Alternatively, run `bin/java -version` after extracting. Replace '+'
+    # symbol in version with '_', otherwise it will be interpreted as a variant
+    version('10.0.1_10', 'ae8ed645e6af38432a56a847597ac61d4283b7536688dbab44ab536199d1e5a4', curl_options=curl_options,
+            url='http://download.oracle.com/otn-pub/java/jdk/10.0.1+10/fb4372174a714e6b8c52526dc134031e/jdk-10.0.1_linux-x64_bin.tar.gz')
+    version('1.8.0_172-b11', 'eda2945e8c02b84adbf78f46c37b71c1', curl_options=curl_options,
             url='http://download.oracle.com/otn-pub/java/jdk/8u172-b11/a58eab1ec242421181065cdc37240b08/jdk-8u172-linux-x64.tar.gz')
-    version('8u141-b15', '8cf4c4e00744bfafc023d770cb65328c', curl_options=curl_options,
+    version('1.8.0_141-b15', '8cf4c4e00744bfafc023d770cb65328c', curl_options=curl_options,
             url='http://download.oracle.com/otn-pub/java/jdk/8u141-b15/336fa29ff2bb4ef291e347e091f7f4a7/jdk-8u141-linux-x64.tar.gz')
-    version('8u131-b11', '75b2cb2249710d822a60f83e28860053', curl_options=curl_options,
+    version('1.8.0_131-b11', '75b2cb2249710d822a60f83e28860053', curl_options=curl_options,
             url='http://download.oracle.com/otn-pub/java/jdk/8u131-b11/d54c1d3a095b4ff2b6607d096fa80163/jdk-8u131-linux-x64.tar.gz')
-    version('8u92-b14',  '65a1cc17ea362453a6e0eb4f13be76e4', curl_options=curl_options)
-    version('8u73-b02',  '1b0120970aa8bc182606a16bf848a686', curl_options=curl_options)
-    version('8u66-b17',  '88f31f3d642c3287134297b8c10e61bf', curl_options=curl_options)
-    # The 7u80 tarball is not readily available from Oracle.  If you have
-    # the tarball, add it to your mirror as mirror/jdk/jdk-7u80.tar.gz and
-    # away you go.
-    version('7u80-b0', '6152f8a7561acf795ca4701daa10a965')
-
-    provides('java@8', when='@8u0:8u999')
-    provides('java@7', when='@7u0:7u999')
-
-    def url_for_version(self, version):
-        url = "http://download.oracle.com/otn-pub/java/jdk/{0}/jdk-{1}-linux-x64.tar.gz"
-        version = str(version)
-        minor_version = version[:version.index('-')]
-        return url.format(version, minor_version)
+    version('1.8.0_92-b14', '65a1cc17ea362453a6e0eb4f13be76e4', curl_options=curl_options)
+    version('1.8.0_73-b02', '1b0120970aa8bc182606a16bf848a686', curl_options=curl_options)
+    version('1.8.0_66-b17', '88f31f3d642c3287134297b8c10e61bf', curl_options=curl_options)
+    version('1.7.0_80-b0', '6152f8a7561acf795ca4701daa10a965', curl_options=curl_options)
```

Comment:
None of these older versions are downloadable, so I removed the `url_for_version` function. Do we even want to keep these versions?